### ISSUE TITLE
Cannot import private key (CLI)

### ISSUE THREAD

**Issue by PSLLSP** [devaultcrypto/devault#273]
DeVault Core RPC client version v1.1.6.0-c687e4a89

I tried DeVault wallet after some time. I cannot import private key that I exported in the past with "dumpprivkey". It looks like method importprivkey was removed. dumpprivkey still works and "help dumprivkey" still describes how to import private key with "importprivkey". This is important issue from my point of view.

Second. The pass phrase method to create new wallet is ugly and buggy (QT wallet). I don't like it. Verification phase of new key is buggy. When I do mistake (word is put to wrong position), in many cases I just cannot undo this error and I have to return one step and start again. The mistake can be made easy, word is dropped to wrong position because user doesn't care too much about precision (or because his mouse is not good) and system just decides that word should be put to other slot. Try it, again and again!. It is time consuming. Copy & paste doesn't work at restore process too, each word has to be manually typed (I cannot paste all 24 words in single copy&paste operation). Maybe this process is easier when English is your native langue that is not my case... I understand that you try to make live of users easier, to remove need to remember long hashes but this attempt failed from my point of view. Old paper wallets were really user friendly but this 12 or 24 pass phrase method is not user friendly...

**Comment by jonspock** [comment]
thanks for the feedback.
Importprivkey was replaced with Sweepprivkey which will move the funds to an address in your wallet that is backed by a word phrase. That way the wallet will always be recoverable.
If there is a good use case for needing 'importprivkey' we can add it back, but it's typically not available for HD only wallets, such as those on mobile.
Can you show us a QT wallet that is less buggy or ugly as a comparison that we can strive to emulate? Otherwise, we'd need more details to understand the issue. 
If you want to cut-and-paste the whole word phrase, I'd advise you to remove old wallet (or start afresh) and use the command line option -seedphrase such as ./devaultd -seedphrase="zoo wrong zoo wrong zoo wrong zoo...."

**Comment by PSLLSP** [comment]
1) devault-qt  -seedphrase="xxx"; it doesn't work. I tried with 12 words and with 24 words, it just doesn't work.

this is from help:
  -seedphrase=<words>
       Specify 12-word seed phrase for wallet on 1st run (only)

Is 24-word phrase (preferred option) supported too? Anyway, I tried with 12-words and it was not loaded, I have to type words manually...

2) Seed phrase  words verification is difficult. Create an error, like swap two first words. Fill all 12-words, and move to the next step. Seed is rejected so go back and try to fix it. You CANNOT fix it (to swap two words), you have to come back and start verification  process from the beginning. Are you boring? Try with 24-words phrase...

3) When new seed phrase is generated, there is no way to just Copy&Paste the sequence to a file. I have to manually type it. This is really user unfriendly, time consuming and errors can be done. Well, I can take screenshot but how to get words from the screenshot? Is (online) OCR safe way?? Other issue is that there is no clear relation between graphical representation (2x6 or 4x6) of seed phrase and "1-line list (1x12 or 1x24)" representation. I assume that mapping is from left to right, than next line etc, like is standard in western culture. Is this assumption correct??

4) wallet synchronization issues. I reported this problem a year ago, I still can replicate it. Wallet can end in a state that it cannot synchronize, it reports header error and puts all peers to banned list. Blocks has to be deleted and the whole block-chain loaded from block zero, it is 5 or 6GB now... :-(
How to replicate this problem? Simulate "low disk space" situation.Wallet will detect that disk is out of free space (200MB free is left?); wallet will stop syncing  and execute controlled shutdown. Unfortunately, once some disk space is free, wallet will not synchronize but it will reject all peers with an error... 

I vote to don't remove "importprivkey" until phrase seed and syncing issues are fixed. Other reason is compatibility with other coins, they still support importprivkey command.

**Comment by PSLLSP** [comment]
One more note. To restore wallet with seed phase, I have to type 12 or 24 words (QT wallet). When I paste several words to the cell, they are not separated. I assume it is not real problem for programmer to detect that several words were pasted and split them and fill following cells... Do it in smart way, when two words are pasted, fill current and the next cell. When 6 words are pasted, fill 6 cells and when 24 words are pasted to the first cell, fill all cells...

**Comment by jonspock** [comment]
If you look at my comment I suggested using "devaultd" not "devault-qt" with -seedphrase.
We've intentionally made it difficult to go back and forth between cut-and-paste of complete phrase between wallet and other text files, since we want people to write down the phrase on paper. Putting the phrase into a text file is a security vulnerability.  If you really want to get it though, you can use "dumpwallet" to a file. And then 'devautld` to recreate.
Will take a look at restoring wallet with seed phrase and pasting into 1 cell if the other devs think this is a good idea.
Lastly, can you please use separate Github issues for each independent issue instead of grouping many disparate things into 1 issue? Otherwise, they may get forgotten about as it's too difficult to keep track of what is fixed or not.

**Comment by PSLLSP** [comment]
Bancor web, that is some service to exchange tokens, allows user to create a wallet with 12 words phrase. Once wallet is generated, words are printed with sequence numbers (so it is clear that sequence is from left to right, than next line), phrase words are printed in grid 3x4. They allow to copy words to clipboard, there is direct "Copy to clipboard" link. I like it!

From my point of view, it is user responsibility to store seed phrase at secure place, it could be a paper (written by hand or printed on printer), papyrus, stone (more reliable than paper or papyrus, it could survive really long time) or just USB drive or secure cloud storage or special device to store cryptocurrency (like TREZOR).

**Comment by PSLLSP** [comment]
One more idea from Bancor web. Phrase verification. Words are just clicked in the sequence and they are automatically added to the end of the list, it starts with the first word, than the second one, etc, When user makes a mistake, it can click on "wrong" word in the list and it is removed. I like it! It is more user friendly GUI implementation than what is in Devault wallet now, where each word has to be dropped with mouse to the right slot (more work, easier to do a mistake, even more difficult on touch screen device).

**Comment by PSLLSP** [comment]
> If you look at my comment I suggested using "devaultd" not "devault-qt" with -seedphrase.

I know that you used headless client but "command line help" (in QT GUI HELP) clearly states that qt client supports command line options.

DeVault Core version v1.1.6.0-c687e4a89 (64-bit)
Usage:  devault-qt [command-line options]                     
...
-seedphrase=<words> | Specify 12-word seed phrase for wallet on 1st run (only)
...

Other point to seed phrase. Is there a command  for headless client to print seedphrase? I know that GUI wallet can display wallet seedphrase and it works. I cannot find similar command for CLI interface. I only found that I can use "dumpwallet" that exports wallet to a text file and  seedphrase is stored in that file too. Maybe there is just an easier way to get just seedphrase from CLI.

**Comment by pppest** [comment]
yes, you can get the seed phrase from `devaultd` by dumping the wallet:
`devault-cli dumpwallet <filename>`

**Comment by jonspock** [comment]
Please see:
https://github.com/devaultcrypto/devault/pull/300
